### PR TITLE
Bump up the csi spec version mention from 0.4.0 to 1.2.0

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

```release-note-none

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ CSI driver.
 
 ### Note
 
-* Master is for CSI v0.4.0. Please see the branches for other CSI releases.
+* Master is for CSI v1.2.0. Please see the branches for other CSI releases.
```
Comment (msau42) [inline README.md:34]: The golang note is also out of date.

What do you think about having a compatibility matrix similar to what the other sidecar repos have?

Comment (humblec) [reply README.md:34]: @msau42 thought about the `golang` , but it mention about minimum supported version ( here `1.9+`), so is it correct to mention that we need `1.12`+ for master ? 

Regarding the compatibility matrix that's a good idea :+1: . However I have to fetch the details somehow. Its also possible to have a followup PR for compatibility matrix.

Comment (msau42) [reply README.md:34]: I'm not sure what the min version of golang is. Our CI uses go1.13. @pohly do you have any thoughts here?

Comment (pohly) [reply README.md:34]: I don't know what the oldest version of Go is that the code still builds with.

I suppose a correct statement would be "building has been tested with the Go version specified in release-tools/travis.yml"...

Comment (humblec) [reply README.md:34]: Yeah, that would be better, confirming the golang compatibility will be a tediuous task unless we have building infra with all the different versions of `go`. I have updated the PR to include above statement. PTAL @pohly @msau42

**Pull request merged**